examples/demo_dlmbl/python/solution.py

```diff
@@ -0,0 +1,506 @@
+batch = next(iter(train_dataloader))
+log_batch_tensorboard(batch, 0, writer, "augmentation/none")
+
+for i, batch in enumerate(train_dataloader):
+    log_batch_tensorboard(batch, i, writer, "augmentation/none")
+writer.close()
+
+# %% Use the following if you need to bring up the tensorboard session again
+# notebook.list()
+# notebook.display(port=6006, height=800)
+
+# %% [markdown]
+"""
+## View augmentations using tensorboard.
+
+<div class="alert alert-info">
+Task 1.3
+Turn on augmentation and view the batch in tensorboard.
+"""
+# %% 
+# data_module.augment = ...
+# data_module.batch_size = ...
+# ... # Feel free to adjust a few other parameters of data_module.
+# data_module.setup("fit")
+# train_dataloader = data_module.train_dataloader()
+# # Draw batches and write to tensorboard
+# writer = SummaryWriter(log_dir = f"{log_dir}/view_batch")
+# for i, batch in enumerate(train_dataloader):
+#     log_batch_tensorboard(...)
+# writer.close()
+
+# %% tags=["solution"]
+data_module.augment = True
+data_module.batch_size = 21
+data_module.split_ratio = 0.8
+data_module.setup("fit")
+
+train_dataloader = data_module.train_dataloader()
+# Draw batches and write to tensorboard
+writer = SummaryWriter(log_dir = f"{log_dir}/view_batch")
+for i, batch in enumerate(train_dataloader):
+    log_batch_tensorboard(batch, i, writer, "augmentation/some")
+writer.close()
+
+# %% [markdown]
+"""
+##  Construct a 2D U-Net for image translation.
+See ``viscy.unet.networks.Unet2D.Unet2d`` for configuration details.
+We setup a fresh data module and instantiate the trainer class.
+"""
+
+# %% The entire training loop is contained in this block.
+
+GPU_ID = 0
+BATCH_SIZE = 10
+
+
+# Dictionary that specifies key parameters of the model.
+phase2fluor_config = {
+    "architecture": "2D",
+    "num_filters": [24, 48, 96, 192, 384],
+    "in_channels": 1,
+    "out_channels": 2,
+    "residual": True,
+    "dropout": 0.1, # dropout randomly turns off weights to avoid overfitting of the model to data.
+    "task": "reg", # reg = regression task.
+}
+
+phase2fluor_model = VSUNet(
+    model_config=phase2fluor_config.copy(),
+    batch_size=BATCH_SIZE,
+    loss_function=torch.nn.functional.l1_loss,
+    schedule="WarmupCosine",
+    log_num_samples=10,
+)
+
+# Reinitialize the data module. 
+phase2fluor_data = HCSDataModule(
+    data_path,
+    source_channel="Phase",
+    target_channel=["Nuclei", "Membrane"],
+    z_window_size=1,
+    split_ratio=0.8,
+    batch_size=BATCH_SIZE,
+    num_workers=8,
+    architecture="2D",
+    yx_patch_size=(512, 512), 
+    augment = True 
+)
+phase2fluor_data.setup("fit")
+# fast_dev_run runs a single batch of data through the model to check for errors.
+trainer = VSTrainer(accelerator="gpu", devices=[GPU_ID], max_epochs = 40,  fast_dev_run=True)
+
+# trainer class takes the model and the data module as inputs.
+trainer.fit(phase2fluor_model, datamodule=phase2fluor_data)
+
+# %% [markdown]
```

Comment:
@ziw-liu how do you typically log graphs?

Reply:
The lightning trainer logs graph automatically when `model.example_input_array` is available. It should be possible to manually do `logger.log_graph(model, array)` as well.

Reply:
> It should be possible to manually do `logger.log_graph(model, array)` as well.

It won't work here though, as logging is completely disabled when `fast_dev_run=True`.

Reply:
https://github.com/mehta-lab/VisCy/pull/36/commits/9d29413a3b7e28982d093db2c0f9917a4445e83f seems to fix this.